src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageReadmeControl.xaml.cs

```diff
@@ -17,54 +20,108 @@ namespace NuGet.PackageManagement.UI
     /// <summary>
     /// Interaction logic for PackageReadmeControl.xaml
     /// </summary>
-    public partial class PackageReadmeControl : UserControl
+    public partial class PackageReadmeControl : UserControl, IDisposable, INotifyPropertyChanged
     {
 #pragma warning disable CS0618 // Type or member is obsolete
         private IMarkdownPreview _markdownPreview;
 #pragma warning restore CS0618 // Type or member is obsolete
+        private bool _isDisposed = false;
+        private CancellationTokenSource _controlDisposingTokenSource;
+        private bool _isBusy = true;
 
         public PackageReadmeControl()
         {
             InitializeComponent();
-            _markdownPreview = MarkdownPreviewSingleton.GetInstance();
-            descriptionMarkdownPreview.Content = _markdownPreview.VisualElement;
+            _controlDisposingTokenSource = new CancellationTokenSource();
+        }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        /// <summary>
+        /// Added to allow the view to display the loading spinner while the markdown is being rendered 
```

Comment:
I don't think the comment is necessary. It's clear to me what the property is used for, but I'm unclear why it's here and not in the ViewModel.